packages/adaptive_scaffold/lib/src/adaptive_scaffold.dart

```diff
@@ -0,0 +1,469 @@
+  final int? selectedIndex;
+
+  /// Widget to be displayed in the body slot at the smallest breakpoint.
+  ///
+  /// If nothing is entered for this property, then the default [body] is
+  /// displayed in the slot. If null is entered for this slot, the slot stays
+  /// empty.
+  final WidgetBuilder? smallBody;
+
+  /// Widget to be displayed in the body slot at the middle breakpoint.
+  ///
+  /// The default displayed body.
+  final WidgetBuilder? body;
+
+  /// Widget to be displayed in the body slot at the largest breakpoint.
+  ///
+  /// If nothing is entered for this property, then the default [body] is
+  /// displayed in the slot. If null is entered for this slot, the slot stays
+  /// empty.
+  final WidgetBuilder? largeBody;
+
+  /// Widget to be displayed in the secondaryBody slot at the smallest
+  /// breakpoint.
+  ///
+  /// If nothing is entered for this property, then the default [secondaryBody]
+  /// is displayed in the slot. If null is entered for this slot, the slot stays
+  /// empty.
+  final WidgetBuilder? smallSecondaryBody;
+
+  /// Widget to be displayed in the secondaryBody slot at the middle breakpoint.
+  ///
+  /// The default displayed secondaryBody.
+  final WidgetBuilder? secondaryBody;
+
+  /// Widget to be displayed in the seconaryBody slot at the smallest
+  /// breakpoint.
+  ///
+  /// If nothing is entered for this property, then the default [secondaryBody]
+  /// is displayed in the slot. If null is entered for this slot, the slot stays
+  /// empty.
+  final WidgetBuilder? largeSecondaryBody;
+
+  /// Defines the fractional ratio of body to the secondaryBody.
+  ///
+  /// For example 0.3 would mean body takes up 30% of the available space and
+  /// secondaryBody takes up the rest.
+  ///
+  /// If this value is null, the ratio is defined so that the split axis is in
+  /// the center of the screen.
+  final double? bodyRatio;
+
+  /// The breakpoint defined for the small size, associated with mobile-like
+  /// features.
+  ///
+  /// Defaults to [Breakpoints.small].
+  final Breakpoint smallBreakpoint;
+
+  /// The breakpoint defined for the medium size, associated with tablet-like
+  /// features.
+  ///
+  /// Defaults to [Breakpoints.mediumBreakpoint].
+  final Breakpoint mediumBreakpoint;
+
+  /// The breakpoint defined for the large size, associated with desktop-like
+  /// features.
+  ///
+  /// Defaults to [Breakpoints.largeBreakpoint].
+  final Breakpoint largeBreakpoint;
+
+  /// Whether or not the developer wants the smooth entering slide transition on
+  /// secondaryBody.
+  ///
+  /// Defaults to true.
+  final bool internalAnimations;
+
+  /// Whether to orient the body and secondaryBody in horizontal order (true) or
+  /// in vertical order (false).
+  ///
+  /// Defaults to true.
+  final bool horizontalBody;
+
+  /// Whether to use a [Drawer] over a [BottomNavigationBar] when not on mobile
+  /// and Breakpoint is small.
+  ///
+  /// Defaults to true.
+  final bool useDrawer;
```

Comment:
I think this could be a more descriptive name, since it seems like it only applies to desktop.  Maybe `preferDrawerOnDesktop`?

Couldn't this instead be an attribute of the breakpoint?

Reply:
It is actually an attribute of the breakpoint. I just added this option as a way to override that. One of the properties is drawerBreakpoint which defaults to smallDesktop and that is the breakpoint to be used to decide when to show the drawer. I believe drawers are also used on mobile so I had it as a generic `userDrawer` for that reason but I see why that might not be the best naming.